### DISCUSSION TITLE
Overlapping routes with `nest("/", _)`

### DISCUSSION THREAD

**Discussion by wooden-worm** [tokio-rs/axum#453] [Q&A] [answered]
I wanna port an existing web app to axum, but having problems with overlapping routes. The problem is I wanna serve static files under `/`. The routing logic is essentially:
- Check for dynamic content first, if not
- Check for the static file

I can't use `.nest("/", get(static_file_svc)).nest("/app", app::routes())` due to non-overlapping route rule.

My questions are:
- Is this something `axum` will support in the future?
- I think I can work around the problem by defining an catch all route `.nest("/", my_svc)`, and in `my_svc` manually implement the above routing logic. Is this gonna work, considering the dynamic routes have to handle both `get` and `post` requests?

**Comment by davidpdrsn** [comment, marked as answer]
This exact use case was mentioned in the [changelog](https://github.com/tokio-rs/axum/blob/main/CHANGELOG.md) actually:

>   - **added:** `Router::fallback` for adding handlers for request that didn't match any routes. `Router::fallback` must be use instead of `nest("/", _)` ([#408])

So this should work `.fallback(static_file_svc).nest("/app", app::routes())`. The order doesn't matter. You might also wanna update to `static_file_svc` to return some meaningful response for non-GET requests, probably 404.

**Reply by wooden-worm** [reply]
Ah that's perfect, thank you!
